cmd/kubeadm/app/phases/controlplane/manifests_test.go

```diff
@@ -807,74 +953,56 @@ func TestGetSchedulerCommand(t *testing.T) {
 	}
 }
 
-func TestGetAuthzParameters(t *testing.T) {
+func TestGetAuthzModes(t *testing.T) {
 	var tests = []struct {
+		name     string
 		authMode []string
-		expected []string
+		expected string
 	}{
 		{
+			name:     "default if empty",
 			authMode: []string{},
-			expected: []string{
-				"--authorization-mode=Node,RBAC",
-			},
+			expected: "Node,RBAC",
 		},
 		{
-			authMode: []string{"RBAC"},
-			expected: []string{
-				"--authorization-mode=RBAC",
-			},
+			name:     "add missing Node",
+			authMode: []string{authzmodes.ModeRBAC},
+			expected: "Node,RBAC",
 		},
 		{
-			authMode: []string{"AlwaysAllow"},
-			expected: []string{
-				"--authorization-mode=AlwaysAllow",
-			},
+			name:     "add missing RBAC",
+			authMode: []string{authzmodes.ModeNode},
+			expected: "Node,RBAC",
 		},
 		{
-			authMode: []string{"AlwaysDeny"},
-			expected: []string{
-				"--authorization-mode=AlwaysDeny",
-			},
+			name:     "add defaults to ABAC",
+			authMode: []string{authzmodes.ModeABAC},
+			expected: "ABAC,Node,RBAC",
 		},
 		{
-			authMode: []string{"ABAC"},
-			expected: []string{
-				"--authorization-mode=ABAC",
-				"--authorization-policy-file=/etc/kubernetes/abac_policy.json",
-			},
+			name:     "add default to Webhook",
+			authMode: []string{authzmodes.ModeWebhook},
+			expected: "Node,RBAC,Webhook",
 		},
 		{
-			authMode: []string{"ABAC", "Webhook"},
-			expected: []string{
-				"--authorization-mode=ABAC,Webhook",
-				"--authorization-policy-file=/etc/kubernetes/abac_policy.json",
-				"--authorization-webhook-config-file=/etc/kubernetes/webhook_authz.conf",
-			},
-		},
-		{
-			authMode: []string{"ABAC", "RBAC", "Webhook"},
-			expected: []string{
-				"--authorization-mode=ABAC,RBAC,Webhook",
-				"--authorization-policy-file=/etc/kubernetes/abac_policy.json",
-				"--authorization-webhook-config-file=/etc/kubernetes/webhook_authz.conf",
-			},
-		},
-		{
-			authMode: []string{"Node", "RBAC", "Webhook", "ABAC"},
-			expected: []string{
-				"--authorization-mode=Node,RBAC,Webhook,ABAC",
-				"--authorization-policy-file=/etc/kubernetes/abac_policy.json",
-				"--authorization-webhook-config-file=/etc/kubernetes/webhook_authz.conf",
-			},
+			name:     "all",
```

Comment:
also add a test case for `RBAC,Webhook` => `Node,RBAC,Webhook`

Reply:
please address this still, and add some unit tests for `AlwaysAllow/AlwaysDeny/{and some misc random values}`